specifications_dbus.md

```diff
@@ -0,0 +1,92 @@
+# Specifications
+
+## Index
+
+* [Connection](#connection)
+* [Connector](#connector)
+* [Connector API](#connector-API)
+* [Distributor](#distributor)
+* [Distributor API](#distributor-api)
+* [Appendix: Implementation practices](#appendix-implementation-practices)
+
+## Connection
+
+A *connection* is created when a connector registers with a distributor. It is uniquely identified by its *token*, a random string which the connector generates as part of the registration process.
+
+## Connector
+
+A *connector* is an application, or software library used in an application, which wants to receive push messages from a UnifiedPush distributor.
+
+When a connector first wants to register, it should fetch all service names currently registered on the session bus, and filter it to only those which begin with `org.unifiedpush.Distributor.`. It should then choose one of these (see [Appendix: Implementation practices](#appendix-implementation-practices)) to use as its distributor. It must then generate a random string to use as a token, and call the [org.unifiedpush.Distributor1.Register](#orgunifiedpushdistributor1register-string--string-string) method with this token as the argument. If the return value indicates "NEW_ENDPOINT", the connector must then save the token and the endpoint received in its persistent storage. Registration is now complete.
+
+On subsequent startups, the connector should call [org.unifiedpush.Distributor1.Register](#orgunifiedpushdistributor1register-string--string-string) with the *same* token, and store the new endpoint returned, if it has changed. The connector should also implement the [org.unifiedpush.Connector1.NewEndpoint](#orgunifiedpushconnector1newendpoint-string-string--nothing) method to receive notification about a changed endpoint from the distributor at any time.
+
+The application SHOULD provide an activatable D-Bus service, so that it can be activated by a new push message when it is not running. See [Appendix: Implementation practices](#appendix-implementation-practices).
```

Comment:
SHOULD or MUST ? Actually, foreground notifications are mostly handled by the app itself. Moreover, it is usual to see app ignoring incoming notif when it is in a foreground state. For convenience, we may prefer a MUST there, even if it is technically possible to do without.

Reply:
I only said SHOULD because it shouldn't be a hard error to *not* have an activatable service (sure then you won't receive notifications when not running, but I thought of that more as a problem for the implementation to solve rather than the protocol to demand).